### PR TITLE
Add request-reply use case [16609]

### PR DESCRIPTION
**Pull request by richiware**
This PR adds a new use case: Request-Replyg communication.

Depends on:
- eProsima/Fast-DDS#3173

### CODE REVIEW

File: docs/fastdds/use_cases/request_reply/request_reply.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+it has to retrieve the request's identifier using |SampleInfo::sample_identity-api|.
+
+.. literalinclude:: ../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 4
+   :start-after: //REQUEST_REPLY_EXAMPLE_SERVER_GET_ID
+   :end-before: //!
+
+After processing the request, the server should send the reply and it have to be associated with the requestt assigning
+the stored identifier in |WriteParams-related_sample_identity-api|.
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/request_reply/request_reply.rst:133]:
```suggestion
After processing the request, the server should send the reply to the client with the related request attached.
This is done assigning the stored identifier in |WriteParams-related_sample_identity-api|.
```

File: docs/fastdds/use_cases/request_reply/request_reply.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+   :language: c++
+   :dedent: 4
+   :start-after: //REQUEST_REPLY_EXAMPLE_SERVER_SEND_REPLY
+   :end-before: //!
+
+Identifying the reply for the client
+------------------------------------
+
+When the client application receives a reply (for example through |DataReaderListener::on_data_available-api|),
+the client application should identify which reply is an answer for its request.
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/request_reply/request_reply.rst:145]:
```suggestion
the client application should identify if the reply received is the one expected for its request.
```

File: docs/fastdds/use_cases/request_reply/request_reply.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+For this the client application have to compare the stored |SampleIdentity-api| with the incomming
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/request_reply/request_reply.rst:146]:
```suggestion
For this the client application has to compare the stored |SampleIdentity-api| with the incoming
```

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
A few nitpicks

File: docs/fastdds/use_cases/request_reply/request_reply.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+           long x;
+           long y;
+       };
+
+       struct ReplyType
+       {
+           long long z;
+       };
+
+2. In the client application creates a DataWriter for the request and a DataReader for the reply.
```
[the remark this answers is not in the document]
Comment (richiware) [reply docs/fastdds/use_cases/request_reply/request_reply.rst]: Done in ba11b18e

File: docs/fastdds/use_cases/request_reply/request_reply.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+    .. literalinclude:: ../../../../code/DDSCodeTester.cpp
+       :language: c++
+       :dedent: 4
+       :start-after: //REQUEST_REPLY_EXAMPLE_CLIENT_CREATE_ENTITIES
+       :end-before: //!
+
+3. In the server application creates a DataWriter for the reply and a DataReader for the request.
```
[the remark this answers is not in the document]
Comment (richiware) [reply docs/fastdds/use_cases/request_reply/request_reply.rst]: Done in ba11b18e

File: docs/fastdds/use_cases/request_reply/request_reply.rst (hunk 2 above)
Comment (richiware) [reply docs/fastdds/use_cases/request_reply/request_reply.rst]: Done in ba11b18e

File: code/DDSCodeTester.cpp [hunk 6]
```diff
@@ -5541,6 +5541,150 @@ void dds_zero_copy_example()
...
+    Subscriber* subscriber = participant->create_subscriber(eprosima::fastdds::dds::SUBSCRIBER_QOS_DEFAULT);
+
+    //REQUEST_REPLY_EXAMPLE_CLIENT_CREATE_ENTITIES
+    participant->register_type(request_type);
+    participant->register_type(reply_type);
+
+    Topic* request_topic = participant->create_topic("CalculatorRequest",
+                    request_type.get_type_name(), TOPIC_QOS_DEFAULT);
+
+    Topic* reply_topic = participant->create_topic("CalculatorReply", reply_type.get_type_name(), TOPIC_QOS_DEFAULT);
```
[the remark this answers is not in the document]
Comment (richiware) [reply code/DDSCodeTester.cpp:5601]: I think there is a significant difference that has to be shown: in one application the request topic is for the datareader and in the other one for the datawriter, and the same in reply topic.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM!

**Pull request merged**